mm2src/mm2_main/src/lp_swap.rs

```diff
@@ -1415,6 +1497,137 @@ pub struct SwapPubkeys {
+    interval_sec: f64,
+    p2p_privkey: Option<KeyPair>,
+) -> AbortOnDropHandle {
+    let fut = async move {
+        loop {
+            broadcast_swap_v2_message(&ctx, topic.clone(), &msg, &p2p_privkey);
+            Timer::sleep(interval_sec).await;
+        }
+    };
+    spawn_abortable(fut)
+}
+
+/// Processes messages received during execution of the upgraded swap protocol.
+pub fn process_swap_v2_msg(ctx: MmArc, topic: &str, msg: &[u8]) -> P2PProcessResult<()> {
+    use prost::Message;
+
+    let uuid = Uuid::from_str(topic).map_to_mm(|e| P2PProcessError::DecodeError(e.to_string()))?;
+
+    let swap_ctx = SwapsContext::from_ctx(&ctx).unwrap();
+    let mut msgs = swap_ctx.swap_v2_msgs.lock().unwrap();
+    if let Some(msg_store) = msgs.get_mut(&uuid) {
+        let signed_message = SignedMessage::decode(msg).map_to_mm(|e| P2PProcessError::DecodeError(e.to_string()))?;
+
+        let pubkey =
+            PublicKey::from_slice(&signed_message.from).map_to_mm(|e| P2PProcessError::DecodeError(e.to_string()))?;
+        let signature = Signature::from_compact(&signed_message.signature)
+            .map_to_mm(|e| P2PProcessError::DecodeError(e.to_string()))?;
+        let secp_message = secp256k1::Message::from_slice(sha256(&signed_message.payload).as_slice())
+            .expect("sha256 is 32 bytes hash");
+
+        SECP_VERIFY
+            .verify(&secp_message, &signature, &pubkey)
+            .map_to_mm(|e| P2PProcessError::InvalidSignature(e.to_string()))?;
+
+        let swap_message = SwapMessage::decode(signed_message.payload.as_slice())
+            .map_to_mm(|e| P2PProcessError::DecodeError(e.to_string()))?;
+
+        debug!("Processing swap v2 msg {:?} for uuid {}", swap_message, uuid);
+        match swap_message.inner {
+            Some(swap_v2_pb::swap_message::Inner::MakerNegotiation(maker_negotiation)) => {
+                msg_store.maker_negotiation = Some(maker_negotiation)
+            },
+            Some(swap_v2_pb::swap_message::Inner::TakerNegotiation(taker_negotiation)) => {
+                msg_store.taker_negotiation = Some(taker_negotiation)
+            },
+            Some(swap_v2_pb::swap_message::Inner::MakerNegotiated(maker_negotiated)) => {
+                msg_store.maker_negotiated = Some(maker_negotiated)
+            },
+            Some(swap_v2_pb::swap_message::Inner::TakerFundingInfo(taker_funding)) => {
+                msg_store.taker_funding = Some(taker_funding)
+            },
+            Some(swap_v2_pb::swap_message::Inner::MakerPaymentInfo(maker_payment)) => {
+                msg_store.maker_payment = Some(maker_payment)
+            },
+            Some(swap_v2_pb::swap_message::Inner::TakerPaymentInfo(taker_payment)) => {
+                msg_store.taker_payment = Some(taker_payment)
+            },
+            Some(swap_v2_pb::swap_message::Inner::TakerPaymentSpendPreimage(preimage)) => {
+                msg_store.taker_payment_spend_preimage = Some(preimage)
+            },
```

Comment:
I believe this will need a `Some(_)` case in this match statement. I will research it further after finishing static analysis of the rest of this PR. 

I expect this will act unexpectedly, possibly crashing the thread, if a malicious node sends a messages with `Some(an_unexpected_type)`.

Having a catch all case for `Some(_)` will also allow for better backwards compatibility if we ever had to add additional message types.

Reply:
Will do some further testing once finished with review. I have reconsidered this, and I believe libp2p networking would not allow for mismatching types(eg, a v1 vs a v2)